# Always use local DevTools files instead of remote files from Google
# NOTE: This can break Remote Debugging
# This also fixes local debugging with domain substitution
# Related comment: https://bugs.chromium.org/p/chromium/issues/detail?id=710701#c14

--- a/third_party/devtools-frontend/src/BUILD.gn
+++ b/third_party/devtools-frontend/src/BUILD.gn
@@ -209,14 +209,13 @@ generated_non_autostart_non_remote_modul
   "$resources_out_dir/web_audio/web_audio_module.js",
   "$resources_out_dir/webauthn/webauthn_module.js",
   "$resources_out_dir/workspace_diff/workspace_diff_module.js",
-]
-
-generated_remote_modules = [
   "$resources_out_dir/lighthouse_worker/lighthouse_worker_module.js",
   "$resources_out_dir/dagre_layout/dagre_layout_module.js",
   "$resources_out_dir/emulated_devices/emulated_devices_module.js",
 ]
 
+generated_remote_modules = []
+
 generated_test_modules = [
   "$resources_out_dir/accessibility_test_runner/accessibility_test_runner_module.js",
   "$resources_out_dir/application_test_runner/application_test_runner_module.js",
@@ -273,7 +272,7 @@ devtools_frontend_resources_deps = [
   ":copy_emulated_devices_images",
   ":devtools_extension_api",
   "front_end",
-  "test",
+#  "test",
 ]
 
 if (is_debug) {
@@ -337,7 +336,6 @@ action("generate_devtools_grd") {
       generated_typescript_modules +
       [
         "$resources_out_dir/devtools_extension_api.js",
-        "$resources_out_dir/sdk/wasm_source_map/pkg/wasm_source_map_bg.wasm",
         "$resources_out_dir/formatter_worker_entrypoint.js",
         "$resources_out_dir/third_party/acorn/package/dist/acorn.mjs",
       ]
--- a/third_party/devtools-frontend/src/devtools_grd_files.gni
+++ b/third_party/devtools-frontend/src/devtools_grd_files.gni
@@ -114,6 +114,7 @@ grd_files_release_sources = [
   "front_end/css_overview/css_overview-legacy.js",
   "front_end/css_overview/css_overview.js",
   "front_end/css_overview/css_overview_module.js",
+  "front_end/dagre_layout/dagre_layout_module.js",
   "front_end/data_grid/data_grid-legacy.js",
   "front_end/data_grid/data_grid.js",
   "front_end/data_grid/data_grid_module.js",
@@ -133,6 +134,7 @@ grd_files_release_sources = [
   "front_end/elements/elements-legacy.js",
   "front_end/elements/elements.js",
   "front_end/elements/elements_module.js",
+  "front_end/emulated_devices/emulated_devices_module.js",
   "front_end/emulation/emulation-legacy.js",
   "front_end/emulation/emulation.js",
   "front_end/event_listeners/event_listeners-legacy.js",
@@ -191,6 +193,7 @@ grd_files_release_sources = [
   "front_end/lighthouse/lighthouse.js",
   "front_end/lighthouse/lighthouse_module.js",
   "front_end/lighthouse_worker.js",
+  "front_end/lighthouse_worker/lighthouse_worker_module.js",
   "front_end/main/main-legacy.js",
   "front_end/main/main.js",
   "front_end/marked/marked.js",
@@ -242,7 +245,6 @@ grd_files_release_sources = [
   "front_end/screencast/screencast.js",
   "front_end/sdk/sdk-legacy.js",
   "front_end/sdk/sdk.js",
-  "front_end/sdk/wasm_source_map/pkg/wasm_source_map_bg.wasm",
   "front_end/search/search-legacy.js",
   "front_end/search/search.js",
   "front_end/search/search_module.js",
--- a/third_party/devtools-frontend/src/front_end/devtools_app.json
+++ b/third_party/devtools-frontend/src/front_end/devtools_app.json
@@ -10,10 +10,10 @@
     { "name": "browser_debugger" },
     { "name": "css_overview" },
     { "name": "cookie_table" },
-    { "name": "dagre_layout", "type": "remote" },
+    { "name": "dagre_layout" },
     { "name": "devices" },
     { "name": "elements" },
-    { "name": "emulated_devices" , "type": "remote" },
+    { "name": "emulated_devices" },
     { "name": "issues" },
     { "name": "developer_resources" },
     { "name": "har_importer" },
--- a/third_party/devtools-frontend/src/front_end/lighthouse_worker.json
+++ b/third_party/devtools-frontend/src/front_end/lighthouse_worker.json
@@ -2,6 +2,6 @@
     "worker": true,
     "modules": [
         { "name": "worker_service", "type": "autostart" },
-        { "name": "lighthouse_worker", "type": "remote" }
+        { "name": "lighthouse_worker" }
     ]
 }
--- a/third_party/devtools-frontend/src/front_end/sdk/wasm_source_map/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/sdk/wasm_source_map/BUILD.gn
@@ -8,7 +8,6 @@ devtools_pre_built("wasm_source_map") {
   sources = [
     "pkg/wasm_source_map.d.ts",
     "pkg/wasm_source_map.js",
-    "pkg/wasm_source_map_bg.wasm",
     "wasm_source_map-tsconfig.json",
   ]
 }
